engine/consensus/dkg/reactor_engine.go

```diff
@@ -386,10 +387,21 @@ func (e *ReactorEngine) end(epochCounter uint64) func() error {
 
 		privateShare, _, _ := e.controller.GetArtifacts()
 
-		err = e.dkgState.InsertMyBeaconPrivateKey(epochCounter, privateShare)
-		if err != nil {
-			return fmt.Errorf("couldn't save DKG private key in db: %w", err)
+		if privateShare == nil {
+			// we failed to compute a private key - set the corresponding DKG failure state
+			e.log.Warn().Msg("computed nil private key share")
+			err = e.dkgState.SetDKGEndState(nextEpochCounter, flow.DKGEndStateNoKey)
+			if err != nil {
+				return fmt.Errorf("with nil key could not set dkg end state to %s: %w", flow.DKGEndStateNoKey, err)
+			}
```

Comment:
We get to this case only when there was a `crypto.IsDKGFailureError(err)` in line 378. This means there is no need to update the end state with `DKGEndStateNoKey`.

Reply:
Addressing this in https://github.com/onflow/flow-go/pull/1732